tests/test_smoke.py

```diff
@@ -2194,13 +2276,16 @@ def test_nonexistent_bucket(self, nonexist_bucket_url):
             storage_obj = storage_lib.Storage(source=nonexist_bucket_url.format(
                 random_name=nonexist_bucket_name))
 
-    @pytest.mark.parametrize('private_bucket',
-                             [f's3://imagenet', f'gs://imagenet'])
+    @pytest.mark.parametrize(
+        'private_bucket',
+        [f's3://imagenet', f'gs://imagenet', 'cos://us-east/bucket1'])
     def test_private_bucket(self, private_bucket):
         # Attempts to access private buckets not belonging to the user.
         # These buckets are known to be private, but may need to be updated if
         # they are removed by their owners.
-        private_bucket_name = urllib.parse.urlsplit(private_bucket).netloc
+        private_bucket_name = urllib.parse.urlsplit(private_bucket).netloc if \
+              urllib.parse.urlsplit(private_bucket).scheme != 'cos' else \
+                  urllib.parse.urlsplit(private_bucket).path.strip('/')
```

Comment:
```suggestion
        if private_bucket.startswith('cos'):
              private_bucket_name = urllib.parse.urlsplit(private_bucket).path.strip('/')
        else:
              private_bucket_name = urllib.parse.urlsplit(private_bucket).netloc 
```
nit: for readability